wirelessengine.py

```diff
@@ -648,58 +647,58 @@ def scanForNetworks(interfaceName, frequency=0, printResults=False):
                 errString = 'Device is busy'
             elif (retCode == WirelessNetwork.ERR_OPNOTPERMITTED):
                 errString = errString + '. Did you run as root?'
-            
+
         return retCode, errString, wirelessNetworks
-        
+
     def getFieldValue(p, curLine):
         matchobj = p.search(curLine)
-        
+
         if not matchobj:
             return ""
-            
+
         try:
             retVal = matchobj.group(1)
         except:
             retVal = ""
-            
+
         return retVal
-        
-    def parseIWoutput(iwOutput):
-        
-        # Define search regexes once:
-        p_bss = re.compile('^BSS (.*?)\(')
-        p_ssid = re.compile('^.+?SSID: +(.*)')
-        p_ess = re.compile('^	capability:.*(ESS)')
-        p_ess_privacy = re.compile('^	capability:.*(ESS Privacy)')
-        p_ibss = re.compile('^	capability:.*(IBSS)')
-        p_ibss_privacy = re.compile('^	capability:.*(IBSS Privacy)')
-        p_auth_suites = re.compile('.*?Authentication suites: *(.*)')
-        p_pw_ciphers = re.compile('.*?Pairwise ciphers: *(.*)')
-        p_param_channel = re.compile('^.*?DS Parameter set: channel +([0-9]+).*')
-        p_primary_channel = re.compile('^.*?primary channel: +([0-9]+).*')
-        p_freq = re.compile('^.*?freq:.*?([0-9]+).*')
-        p_signal = re.compile('^.*?signal:.*?([\-0-9]+).*?dBm')
-        p_ht = re.compile('.*?HT20/HT40.*')
-        p_bw = re.compile('.*?\\* channel width:.*?([0-9]+) MHz.*')
-        p_secondary = re.compile('^.*?secondary channel offset: *([^ \\t]+).*')
-        p_thirdfreq = re.compile('^.*?center freq segment 1: *([^ \\t]+).*')
-        p_stationcount = re.compile('.*station count: ([0-9]+)')
-        p_utilization = re.compile('.*channel utilisation: ([0-9]+)/255')
+
+    def regexCompiler(self):
```

Comment:
This is where we save cycles.  We create it once and be done with it via instantiation of the Class.